Sources/AblyChat/Rooms.swift

```diff
@@ -1,5 +1,42 @@
+import Ably
+
 public protocol Rooms: AnyObject, Sendable {
-    func get(roomID: String, options: RoomOptions) throws -> any Room
+    func get(roomID: String, options: RoomOptions) async throws -> any Room
     func release(roomID: String) async throws
     var clientOptions: ClientOptions { get }
 }
+
+internal actor DefaultRooms: Rooms {
+    /// Exposed so that we can test it.
+    internal nonisolated let realtime: RealtimeClient
+    internal nonisolated let clientOptions: ClientOptions
+
+    /// The set of rooms, keyed by room ID.
+    private var rooms: [String: DefaultRoom] = [:]
+
+    internal init(realtime: RealtimeClient, clientOptions: ClientOptions) {
+        self.realtime = realtime
+        self.clientOptions = clientOptions
+    }
+
+    internal func get(roomID: String, options: RoomOptions) throws -> any Room {
```

Comment:
forgotten `async`?

Reply:
No — it's not `async` from the point of view of the actor, only those who call it from outside the actor (hence the `Rooms` protocol method needing to be `async`).

Reply:
But adding `async` keyword seems doesn't change anything for compiler.

Reply:
I don't understand your point?

Reply:
I mean if you write `internal func get(roomID: String, options: RoomOptions) async throws -> any Room {` it compiles and tests succeed.

Reply:
What's strange about it? From the point of view of external callers, it was `async` anyway. And from the point of view of the actor, there's nothing to stop you from making an actor-isolated method `async`.